### ISSUE TITLE
Flutter Web Stripe.instance.confirmSetupIntent

### ISSUE THREAD

**Issue by fahadappguy** [flutter-stripe/flutter_stripe#960]
**Describe the bug**
By confirming the SetupIntent method, it creates Intent successfully as displayed in the [video](https://www.veed.io/view/fb555315-6ea7-457e-8a76-ca7b9b60ff25/showcase?sharingWidget=true) But throws the exception "Unimplemented Error". It looks like this is something due to parsing.

**To Reproduce**
Facing exception "Unimplemented Error" When I execute this:
```
final paymentIntent = await Stripe.instance.confirmSetupIntent(
                clientSecret,
                PaymentMethodParams.card(
                  paymentMethodData: PaymentMethodData(),
                  options: PaymentMethodOptions(),
                ),
              );
```
Where
```
CardField(
          enablePostalCode: true,
          postalCodeHintText: 'Postal code',
          countryCode: 'US',
          onCardChanged: (_) {},
        )
```
[Image for Exception details](https://ibb.co/QFcLqcM)
[Video for more details](https://www.veed.io/view/fb555315-6ea7-457e-8a76-ca7b9b60ff25/showcase?sharingWidget=true)

**Expected behavior**
It should return setupIntent instead of this exception.

**Mac**
 - Device: MacBook Pro
 - OS: MacOS Monterey 12.5.1
 - Packages: 
        flutter_stripe: ^5.1.0
        flutter_stripe_web: ^1.3.0
 - Flutter version [e.g. 3.3.0.]

**Comment by fahadappguy** [comment]
I solved this issue here https://github.com/fahadappguy/flutter_stripe.git

For now, you can use the following solution.

dependencies:
  flutter_stripe: ^5.1.0
  flutter_stripe_web: ^1.3.0

dependency_overrides:
  flutter_stripe_web:
    git:
      url: https://github.com/fahadappguy/flutter_stripe.git
      path: packages/stripe_web

**Comment by fahadappguy** [comment]
> ../…/lib/stripe_checkout.dart:19

You are trying to run the example and yes there's such kind of issues. Try with a new project or remove this param "stripeAccountId: stripeAccountId".

**Comment by debjeetdebnath** [comment]
@fahadappguy Thank you for your suggestion as you said to make a new project and run the example it worked but when I click on the checkout button I got this error - **<error: Bad state: Source maps are not done loading.>
    at Object.createErrorWithStack (http://localhost:5000/dart_sdk.js:5674:12)** and I am not able to find where to put the secret key, I have only put stripePublishableKey.
    If you have a Minimum viable product that is running stripe payment on flutter web can you please share it will be really helpful. :)

**Comment by ikallali** [comment]
+1

**Comment by cornwe19** [comment]
@jamesblasco I'm getting an error like this when I try to use `flutter_stripe_web` 2.1.0 to issue a `confirmSetupIntent` from flutter web. I'm pretty closely following the implementation in [the example code](https://github.com/flutter-stripe/flutter_stripe/blob/main/example/lib/screens/others/setup_future_payment_screen.dart#L129).

```JSON
{
  "error": {
    "code": "parameter_unknown",
    "doc_url": "https://stripe.com/docs/error-codes/parameter-unknown",
    "message": "Received unknown parameter: paymentMethod. Did you mean payment_method?",
    "param": "paymentMethod",
    "request_log_url": "***",
    "type": "invalid_request_error"
  }
}
```

Am I doing something wrong or is the SDK serializing payloads improperly?

**Comment by sleewok** [comment]
> @jamesblasco I'm getting an error like this when I try to use `flutter_stripe_web` 2.1.0 to issue a `confirmSetupIntent` from flutter web. I'm pretty closely following the implementation in [the example code](https://github.com/flutter-stripe/flutter_stripe/blob/main/example/lib/screens/others/setup_future_payment_screen.dart#L129).
> 
> ```json
> {
>   "error": {
>     "code": "parameter_unknown",
>     "doc_url": "https://stripe.com/docs/error-codes/parameter-unknown",
>     "message": "Received unknown parameter: paymentMethod. Did you mean payment_method?",
>     "param": "paymentMethod",
>     "request_log_url": "***",
>     "type": "invalid_request_error"
>   }
> }
> ```
> 
> Am I doing something wrong or is the SDK serializing payloads improperly?
> 
> If it's helpful at all, the payload being sent to `/setup_intents/seti_***/confirm` looks something like the following:
> 
> ```
> paymentMethod[runtimeType]: details
> expected_payment_method_type: card
> use_stripe_sdk: true
> key: pk_test_***
> client_secret: seti_***_secret_***
> ```
> 
> I'm guessing `paymentMethod[runtimeType]` is where things are going sideways

I just implemented this and I am having the same error as well. confirmSetupIntent is NOT working. Stripe responds with a "parameter unknown" error. 

When confirming the setupIntent there is not always a payment_method associated. I am adding a card for future onsession usage. The setupintent confirm returns the payment_method id when successful.

**Comment by sleewok** [comment]
@jamesblasco could you point me to the code where I can work on modifying the parameter property name? Or at least attempt to get this working? I spent quite some time trying to figure out where to make changes, but I didn't have any luck. My app testing is blocked right now because of this and I would like to help if possible.

Thanks!

**Comment by remonh87** [comment]
this should be available in flutter stripe web 4.0.1
